client/src/AppBundle/EventSubscriber/ReportSubmittedSubscriber.php

```diff
@@ -16,24 +19,59 @@ class ReportSubmittedSubscriber implements EventSubscriberInterface
     /* @var Mailer */
     private $mailer;
 
-    public function __construct(ReportApi $reportApi, Mailer $mailer)
+    private LoggerInterface $logger;
+
+    /**
+     * @var DateTimeProvider
+     */
+    private DateTimeProvider $dateTimeProvider;
+
+    public function __construct(ReportApi $reportApi, Mailer $mailer, LoggerInterface $logger, DateTimeProvider $dateTimeProvider)
     {
         $this->reportApi = $reportApi;
         $this->mailer = $mailer;
+        $this->logger = $logger;
+        $this->dateTimeProvider = $dateTimeProvider;
     }
 
     public static function getSubscribedEvents()
     {
         return [
-            ReportSubmittedEvent::NAME => 'sendEmail'
+            ReportSubmittedEvent::NAME => 'submitReportEvent'
         ];
     }
 
+    public function submitReportEvent(ReportSubmittedEvent $event)
+    {
+        if ($event->getSubmittedReport()->getUnSubmitDate() !== null) {
+            try {
+                $this->logReportSubmittedEvent($event);
+            } catch (\Exception $e) {
+            }
+        }
+        $this->sendEmail($event);
+    }
+
```

Comment:
We've tended to try to follow SRP with event subscribers to make testing as simple as possible and increase re-use. Could we get this split out to `sendEmail()` and `log()` (or something similar)?

Reply:
Yeah should have done this. Done now.